lib/client/aws_role/impl.go

```diff
@@ -0,0 +1,213 @@
+package aws_role
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Cloud-Foundations/keymaster/lib/paths"
+
+	"github.com/aws/aws-sdk-go-v2/aws/arn"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+)
+
+const toothlessPolicy = `{
+    "Version": "2012-10-17",
+    "Statement": {
+        "Sid": "HandCrafted",
+        "Effect": "Allow",
+        "Action": "sts:GetCallerIdentity",
+        "Resource": "*"
+    }
+}
+`
+
+func parseArn(arnString string) (*arn.ARN, error) {
```

Comment:
add a comment on why are we doing the magic replcement... it not non-obvious

Reply:
Done.